File: tests/test_train.cpp
// Unit tests for the training loop helpers.
#include <gtest/gtest.h>

#include "config.hpp"
#include "train.hpp"

TEST(Train, LambdaScheduleRamps) {
    const double first = era::lambda_at(0, era::MAX_EPOCHS);
    const double middle = era::lambda_at(era::MAX_EPOCHS / 2, era::MAX_EPOCHS);
    const double last = era::lambda_at(era::MAX_EPOCHS - 1, era::MAX_EPOCHS);

    EXPECT_DOUBLE_EQ(first, 0.0);
    EXPECT_GT(middle, 0.0);
    EXPECT_LT(middle, era::LAMBDA_MAX);
    EXPECT_GT(last, 0.99 * era::LAMBDA_MAX);
    EXPECT_LT(first, middle);
    EXPECT_LT(middle, last);
}

File: tests/test_train.py
"""Unit tests for the training loop helpers."""

from era_translator.config import LAMBDA_MAX, MAX_EPOCHS
from era_translator.train import _lambda_at


def test_lambda_schedule_ramps():
    l0 = _lambda_at(0)
    lmid = _lambda_at(MAX_EPOCHS // 2)
    llast = _lambda_at(MAX_EPOCHS - 1)
    assert l0 == 0.0
    assert 0.0 < lmid < LAMBDA_MAX
    assert llast > 0.99 * LAMBDA_MAX
    assert l0 < lmid < llast
